WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -1076,10 +1076,17 @@ private extension NotificationsViewController {
                    scrollPosition: UITableView.ScrollPosition = .none) {
         selectedNotification = notification
 
-        if let indexPath = tableViewHandler.resultsController.indexPath(forObject: notification), indexPath != tableView.indexPathForSelectedRow {
-            DDLogInfo("\(self) \(#function) Selecting row at \(indexPath) for Notification: \(notification.notificationId) (\(notification.type ?? "Unknown type")) - \(notification.title ?? "No title")")
-            tableView.selectRow(at: indexPath, animated: animated, scrollPosition: scrollPosition)
-        }
+        // also ensure that the index path returned from results controller does not have negative row index.
+        // ref: https://github.com/wordpress-mobile/WordPress-iOS/issues/15370
+        guard let indexPath = tableViewHandler.resultsController.indexPath(forObject: notification),
+              indexPath != tableView.indexPathForSelectedRow,
+              indexPath.row < tableView.numberOfRows(inSection: indexPath.section),
+              indexPath.row >= 0 else {
```

Comment:
Nit/Tip: alternative way to test if a value is within a range is using `(0..<n).contains(indexPath.row)`:

```suggestion
              (0..<tableView.numberOfRows(inSection: indexPath.section)).contains(indexPath.row) else {
```

Or `0..<n ~= indexPath.row` (which is equivalent)
```suggestion
              0..<tableView.numberOfRows(inSection: indexPath.section) ~= indexPath.row else {
```

Both notations as well as your existing code are valid and behave the same, so it's mostly code style and up to preferences for conciseness 🙃

Reply:
Thanks @AliSoftware !